k2/torch/csrc/deserialization.cu

```diff
@@ -368,41 +370,51 @@ k2::FsaClass LoadFsa(const std::string &filename) {
     (K2_TORCH_VERSION_MAJOR == 1 && K2_TORCH_VERSION_MINOR >= 9)
   torch::IValue ivalue = torch::jit::readArchiveAndTensors(
       "data", "", "", type_resolver, obj_loader,
-      /*device=*/c10::nullopt, *reader);
+      /*device=*/map_location, *reader);
 
 #else
   torch::IValue ivalue =
       torch::jit::readArchiveAndTensors("data", type_resolver, obj_loader,
-                                        /*device=*/c10::nullopt, *reader);
+                                        /*device=*/map_location, *reader);
```

Comment:
Since we can load file to a particular device, we indeed no need to implement `To(device)` for FsaClass. You could delete them.